app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerV2.kt

```diff
@@ -447,54 +452,69 @@ class BloodPressureEntrySheetControllerV2 @Inject constructor(
         .withLatestFrom(screenChanges)
         .filter { (_, screen) -> screen == DATE_ENTRY }
 
+    val prefilledDateStream = events
+        .ofType<BloodPressureDateToPrefillCalculated>()
+        .map { it.date }
+
     val newBpDataStream = Observables.combineLatest(dateValidations, bpValidations, patientUuidStream)
         .map { (dateResult, bpResult, patientUuid) ->
-          when {
-            dateResult is Valid && bpResult is Success -> SaveBpData.ReadyToCreate(dateResult.parsedDate, bpResult, patientUuid)
-            else -> SaveBpData.NeedsCorrection
+          if (dateResult is Valid && bpResult is Success) {
+            SaveBpData.ReadyToCreate(dateResult.parsedDate, bpResult.systolic, bpResult.diastolic, patientUuid)
+          } else {
+            SaveBpData.NeedsCorrection
           }
         }
 
     val updateBpDataStream = Observables.combineLatest(dateValidations, bpValidations, existingBpUuidStream)
         .map { (dateResult, bpResult, bpUuid) ->
-          when {
-            dateResult is Valid && bpResult is Success -> SaveBpData.ReadyToUpdate(dateResult.parsedDate, bpResult, bpUuid)
-            else -> SaveBpData.NeedsCorrection
+          if (dateResult is Valid && bpResult is Success) {
+            SaveBpData.ReadyToUpdate(dateResult.parsedDate, bpResult.systolic, bpResult.diastolic, bpUuid)
+          } else {
+            SaveBpData.NeedsCorrection
           }
         }
 
+    val toUtcInstant = { date: LocalDate -> date.atTime(OffsetTime.now(utcClock)).toInstant() }
```

Comment:
This method to calculate the UTC instant is incorrect. If the current time is "1 AM IST Jan 1st 2018", and we convert it into an `Instant` using this method, the `Instant` that is generated will be "7:30 PM UTC Jan 1st 2018" when it should be "7:30 PM UTC Dec 31st 2017". I wrote a simple test to indicate the difference and the right approach to generating an `Instant` in UTC here:

```kotlin
@Test
  fun test() {
    val localDate = LocalDate.of(2018, Month.JANUARY, 1)

    // 1 AM IST converted to UTC
    val offsetTime = LocalTime.of(1, 0).atOffset(ZoneOffset.of("+05:30")).withOffsetSameInstant(ZoneOffset.UTC)
    println(offsetTime)
    println(localDate.atTime(offsetTime).toInstant())

    val offsetDateTime = localDate.atTime(1, 0).atOffset(ZoneOffset.of("+05:30"))
    println(offsetDateTime)
    println(offsetDateTime.withOffsetSameInstant(ZoneOffset.UTC).toInstant())
  }
```

Reply:
I've changed it to,

```kotlin
val userTime = LocalTime.now(userClock)
val userDateTime = date.atTime(currentUtcTime)
userDateTime.atZone(UTC).toInstant()
```

I'll extract this to TimeFunctions with your help when you have time.